Initial Commit of Clean Slate

onuro opened the pull request:
This PR introduces the Clean Slate theme, a blank/starter theme in replacement of Blank Canvas, started out with the purpose of using in Pattern Assembler project.

This theme is a stripped down version of TT3 theme, with some more custom theme.json and template parts updates.

I've kept the fonts from TT3, as the use of a fonts library for this theme TBD.

Please let me know what could be improved with this theme, as I'm still checking for things that may be needed to be updated.

Thank you!

onuro committed:
Clean Slate initial commit

The first commit of the Clean Slate theme, a blank/starter theme.

onuro committed:
broken commit process

Initial commit stuck at delta. will re-commit the fonts

onuro committed:
Revert "broken commit process"

This reverts commit e2653f78a845fc4afe6b5b902383a603bed05127.

scruffian commented:
I think if we want a truly clean slate we should remove a lot more - all the patterns for example can go, and quite a few of the templates as well.

onuro committed:
Removal of all patterns

Removed all patterns from the theme

scruffian commented:
What happens if we remove all templates except index?

onuro commented:
> What happens if we remove all templates except index?

Tbh makes sense. @mtias this is also something along the lines of what you've expressed for this right?

bangank36 commented on clean-slate/parts/post-meta.html:
```diff
@@ -0,0 +1,12 @@
+	</div>
+	<!-- /wp:group -->
+</div>
+<!-- /wp:group -->
```
You need a new line here

bangank36 commented on clean-slate/style.css:
```diff
@@ -0,0 +1,16 @@
+Template: 
+Text Domain: clean-slate
+Tags: one-column, custom-colors, custom-menu, custom-logo, editor-style, featured-images, full-site-editing, rtl-language-support, theme-options, threaded-comments, translation-ready, wide-blocks
+*/
```
New line needed

bangank36 commented on clean-slate/templates/single.html:
```diff
@@ -0,0 +1,61 @@
+</div>
+<!-- /wp:group -->
+
+<!-- wp:template-part {"slug":"footer","tagName":"footer"} /-->
```
New line needed

alaczek commented:
I'm also wondering about the inclusion of font assets. What do you think about removing them and reverting to using system fonts?

arthur791004 commented:
Could we have the `home.html` file without any content and use it to display the home page of this theme? cc @mtias

onuro committed:
Missing new lines added

Missing new lines in templates and parts added

scruffian commented on clean-slate/style.css:
```diff
@@ -10,7 +10,7 @@ Requires PHP: 5.7
 License: GNU General Public License v2 or later
 License URI: http://www.gnu.org/licenses/gpl-2.0.html
-Template: 
+Template:
```
```suggestion
```

onuro replied:
@scruffian should I be removing the Template line all together or just remove the space next to the colon?

jffng replied:
Removing it entirely is best. This line should be included only if there is a Parent theme.

onuro committed:
Removal of parent theme line in style.css

Removed "Template:" declaration line in style.css since this is not a child theme.

onuro committed:
Removal of TT3 fonts

Removed all custom fonts bundled with the theme

onuro committed:
Renamed to Blank Canvas

Per the decision on the name, this will be the replacement of Blank Canvas

scruffian commented on blank-canvas-3/theme.json:
```diff
@@ -0,0 +1,360 @@
+			},
+			"link": {
+				"color": {
+					"text": "var(--wp--preset--color--primary)"
```
It might be interesting to use `ref` properties, to reduce the duplication: https://github.com/WordPress/gutenberg/pull/41696

onuro replied:
Hi @scruffian could you elaborate on this? Is do I simply use the following? Because when I did, the theme.json and the theme style breaks. I also don't understand the reason fully even though I've read the PR.

```
"link": {
				"color": {
					"text": { ref: "styles.color.primary" }
				},
```

scruffian replied:
Try using:

```
"color": {
	"text": { ref: "styles.color.text" }
},
```

The reason being that if a user changes their body text color then links will update automatically.

onuro committed:
Remove all templates except index and singular

scruffian commented:
I'm not clear how "blank" we want this to be. The templates still contain a lot of opinion. Do we want to strip it back to just display page content?

jffng commented on blank-canvas-3/theme.json:
```diff
@@ -0,0 +1,360 @@
+{
+	"customTemplates": [
```
I think the `customTemplates` config can be removed since there is no longer a blank template supplied by the theme.

onuro committed:
Removal of CustomTemplates

Removed custom templates declarations from the theme.json file
